cargo-dist/src/backend/installer/homebrew.rs

```diff
@@ -81,3 +81,98 @@ pub(crate) fn write_homebrew_formula(
     LocalAsset::write_new(&script, &info.inner.dest_path)?;
     Ok(())
 }
+
+/// Converts the provided app name into a Ruby class-compatible
+/// string suitable for use as the class in a Homebrew formula.
+// Homebrew implementation is Formulary.class_s:
+// https://github.com/Homebrew/brew/blob/8c7cd3c0fd46f7808e782e40359c19271f950a75/Library/Homebrew/formulary.rb#L447-L453
+pub fn to_class_case(app_name: &str) -> String {
+    if app_name.is_empty() {
+        return app_name.to_owned();
+    }
+
+    let mut out;
+    // First, we uppercase the first character in the string
+    let (first_char, rest) = app_name.split_at(1);
+    out = format!("{}{rest}", first_char.to_uppercase());
+
+    let mut chars = vec![];
+    let mut iter = out.chars();
+    let mut el = iter.next();
+    while el.is_some() {
+        let char = el.unwrap();
+        // -, _ and . are invalid characters in Ruby classes.
+        // Homebrew handles these by stripping them, then uppercasing
+        // the following character
+        match char {
+            '-' | '_' | '.' => {
+                if let Some(next) = iter.next() {
+                    chars.push(next.to_ascii_uppercase());
+                }
+            }
+            _ => chars.push(char),
+        }
+
+        el = iter.next();
+    }
+    out = chars.iter().collect();
+    // Perform two more small replacements
+    out = out.replace('+', "x");
+    out.replace('@', "AT")
```

Comment:
The ruby impl seems a lot more pedantic about this replacement..? (although i don't think @ can show up in a crate name so moot...)

Reply:
both of these replacements could have been done in the loop, no? Arguably this is more readable, tho.

Reply:
Went for a quirks-compatible approach, with a new test that checks how it stacks against Homebrew.